django/thunderstore/api/cyberstorm/tests/test_packages.py

```diff
@@ -0,0 +1,562 @@
+        == "package_cache_key.test.test.test.test.test.True.True.['a', 'b'].['a', 'b'].test.test.last-updated.1.1"
+    )
+
+
+@pytest.mark.django_db
+def test_PackagePreviewAPIView__paginate__index_error() -> None:
+    PackageListingFactory()
+    PackageListingFactory()
+    params = PARAMS_FIXTURE
+    params["page"] = 5
+    view = PackagePreviewAPIView()
+    with pytest.raises(ParseError) as e_info:
+        view.paginate(params=params, queryset=PackageListing.objects.all(), page_size=1)
+    assert str(e_info.value) == "Page index error: no results on requested page"
+
+
+@pytest.mark.django_db
+def test_only_active_packages_are_returned(api_client: APIClient) -> None:
+    PackageListingFactory(package_kwargs={"is_active": False})
+    PackageListingFactory()
+
+    data = __query_api(api_client)
+
+    assert len(data["packages"]) == 1
+
+
+@pytest.mark.django_db
+def test_rejected_packages_are_not_returned(api_client: APIClient) -> None:
+    listing1 = PackageListingFactory(
+        review_status=PackageListingReviewStatus.unreviewed
+    )
+    listing2 = PackageListingFactory(
+        community_=listing1.community, review_status=PackageListingReviewStatus.approved
+    )
+    PackageListingFactory(
+        community_=listing1.community, review_status=PackageListingReviewStatus.rejected
+    )
+
+    data = __query_api(api_client)
+
+    __assert_packages_by_listings(data, [listing2, listing1])
+
+
+@pytest.mark.django_db
+def test_only_approved_packages_are_returned_when_approval_is_required(
+    api_client: APIClient,
+) -> None:
+    listing1 = PackageListingFactory(
+        review_status=PackageListingReviewStatus.unreviewed
+    )
+    listing1.community.require_package_listing_approval = True
+    listing1.community.save()
+    listing2 = PackageListingFactory(
+        community_=listing1.community, review_status=PackageListingReviewStatus.approved
+    )
+    PackageListingFactory(
+        community_=listing1.community, review_status=PackageListingReviewStatus.rejected
+    )
+
+    data = __query_api(api_client)
+
+    __assert_packages_by_listings(data, listing2)
+
+
+@pytest.mark.django_db
+def test_deprecated_packages_are_returned_only_when_requested(
+    api_client: APIClient,
+) -> None:
+    active = PackageListingFactory()
+    PackageListingFactory(
+        community_=active.community, package_kwargs={"is_deprecated": True}
+    )
+
+    data = __query_api(api_client)
+
+    __assert_packages_by_listings(data, active)
+
+    data = __query_api(api_client, "deprecated=on")
+
+    assert len(data["packages"]) == 2
+
+
+@pytest.mark.django_db
+def test_nsfw_packages_are_returned_only_when_requested(api_client: APIClient) -> None:
+    sfw = PackageListingFactory()
+    PackageListingFactory(community_=sfw.community, has_nsfw_content=True)
+
+    data = __query_api(api_client)
+    __assert_packages_by_listings(data, sfw)
+
+    data = __query_api(api_client, "nsfw=true")
+    assert len(data["packages"]) == 2
```

Comment:
Why not use `__assert_packages_by_listings` here too?

Reply:
fixd